services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java

```diff
@@ -381,6 +382,14 @@ public void setNodeSlotAccordingToDBStore(Store store) {
     LOG.info("Node being started from database.");
   }
 
+  @Subscribe
+  public void onImportedBlock(ImportedBlockEvent event) {
+    if (event.getBlock().getSlot().equals(nodeSlot)) {
+      Bytes32 headBlockRoot = this.stateProcessor.processHead();
+      this.eventBus.post(new BroadcastAttestationEvent(headBlockRoot, nodeSlot));
+    }
+  }
```

Comment:
This seems wrong.  We might import two blocks for the same slot and then this would cause us to produce two conflicting attestations and get slashed. We may also do that because this event is fired both here and in `process_slot` - if the `process_slot` fired first and then this block was imported we'd potentially wind up with two different targets for the produced attestations.

edit: I can see this is aiming to meet the first paragraph under https://github.com/ethereum/eth2.0-specs/blob/v0.10.1/specs/phase0/validator.md#attesting  which says we can publish attestations "early" if we already have the block for that slot.  I suspect we should have nice centralised logic for doing that coordination and not split it across multiple classes to be really sure we have it right.  Strongly suspect we should do it as a separate PR (and since it's not a consensus affecting change it can go straight to master).

Reply:
https://pegasys1.atlassian.net/browse/BC-309?atlOrigin=eyJpIjoiZDNmNGY0ZDk3ZmYzNDI2OWE0Yzk5NWRkMDdhNzY0MDMiLCJwIjoiaiJ9

Reply:
Reverted these changes in this PR.